crates/runtime/src/instance/allocator/pooling/memory_pool.rs

```diff
@@ -156,20 +156,21 @@ impl MemoryPool {
         let pkeys = match config.memory_protection_keys {
             MpkEnabled::Auto => {
                 if mpk::is_supported() {
-                    mpk::keys()
+                    mpk::keys(config.max_memory_protection_keys)
                 } else {
                     &[]
                 }
             }
             MpkEnabled::Enable => {
                 if mpk::is_supported() {
-                    mpk::keys()
+                    mpk::keys(config.max_memory_protection_keys)
                 } else {
                     bail!("mpk is disabled on this system")
                 }
             }
             MpkEnabled::Disable => &[],
         };
+        let pkeys = &pkeys[..pkeys.len().min(config.max_memory_protection_keys)];
```

Comment:
I guess I should just move this logic into `keys` if we end up using this approach.